pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -2200,3 +2214,141 @@ func (a *ApplicationRestoreController) processVMResourcesForVMRestore(restore *s
+				Name:       backupLocation.Name,
+			}
+			resourceExport.Spec.Source = *source
+			resourceExport.Spec.Destination = *destination
+			_, err = kdmpShedOps.Instance().CreateResourceExport(resourceExport)
+			if err != nil {
+				logrus.Errorf("failed to create ResourceExport CR[%v/%v]: %v", resourceExport.Namespace, resourceExport.Name, err)
+				return err
+			}
+			// We will update the state of applicationRestoreCr so that includeResources are processed by resourceExport CR.
+			restore.Status.Stage = storkapi.ApplicationRestoreStageIncludeResources
+			restore.Status.LastUpdateTimestamp = metav1.Now()
+			err = a.client.Update(context.TODO(), restore)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+		logrus.Errorf("failed to get restore resourceExport CR: %v", err)
+		// Will retry in the next cycle of reconciler.
+		return nil
+	} else {
+		namespacedName := types.NamespacedName{}
+		namespacedName.Namespace = restore.Namespace
+		namespacedName.Name = restore.Name
+		var message string
+		// Check the status of the resourceExport CR and update it to the application restore CR
+		logrus.Debugf("resource export: %s, status: %s", resourceExport.Name, resourceExport.Status.Status)
+		switch resourceExport.Status.Status {
+		case kdmpapi.ResourceExportStatusFailed:
+			message = fmt.Sprintf("Error applying resources: %v", err)
+			restore.Status.Status = storkapi.ApplicationRestoreStatusFailed
+			restore.Status.Stage = storkapi.ApplicationRestoreStageFinal
+			restore.Status.Reason = message
+			restore.Status.LastUpdateTimestamp = metav1.Now()
+			restore.Status.FinishTimestamp = metav1.Now()
+			err = a.client.Update(context.TODO(), restore)
+			if err != nil {
+				return err
+			}
+			a.recorder.Event(restore,
+				v1.EventTypeWarning,
+				string(storkapi.ApplicationRestoreStatusFailed),
+				message)
+			log.ApplicationRestoreLog(restore).Errorf(message)
+			return err
+		case kdmpapi.ResourceExportStatusSuccessful:
+			// Modify to have subresource level updating
+			includeResourceList := make([]storkapi.ObjectInfo, 0)
+			for _, resource := range resourceExport.Status.Resources {
+				info := storkapi.ObjectInfo{
+					Name:             resource.Name,
+					Namespace:        resource.Namespace,
+					GroupVersionKind: resource.GroupVersionKind,
+				}
+				includeResourceList = append(includeResourceList, info)
+			}
+			if len(includeResourceList) != 0 {
+				restore.Spec.IncludeResources = append(restore.Spec.IncludeResources, includeResourceList...)
+				log.ApplicationRestoreLog(restore).Tracef("%v: resource export CR successful with vm IncludeResourceMap",
+					fn)
+			}
+			//Progress to next stage so that we dont loop here.
+			restore.Status.Stage = storkapi.ApplicationRestoreStageVolumes
+		default:
+			if len(resourceExport.Status.Status) != 0 {
+				log.ApplicationRestoreLog(restore).Errorf("%v: invalid status for resource export: %s, status: %s", fn, resourceExport.Name, resourceExport.Status.Status)
```

Comment:
We have other status in the `ResourceExport` which will default here and we can't call them as invalid state. Words needs rewording.

Reply:
fixed